lib/smart_proxy_remote_execution_ssh/api.rb

```diff
@@ -28,7 +28,9 @@ class Api < ::Sinatra::Base
         ssh_options[:number_of_password_prompts] = 1
 
         socket = nil
-        if env['rack.hijack?']
+        if env['WEBRICK_SOCKET']
+          socket = env['WEBRICK_SOCKET']
```

Comment:
Is that enough?  Wont Webrick still try to use that socket and send some headers?  The socket should be closed at that point and maybe that is indeed good enough.

Reply:
Have not found a failing situation yet.